python/rikai/contrib/torch/inspect/ssd.py

```diff
@@ -0,0 +1,191 @@
+#  Copyright 2022 Rikai Authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""Inspect detail of Torchvision SSD model
```

Comment:
1. I wonder if it's better to organize things as `[torch/transforms/ssd.py, torch/inspect/ssd.py]` or `[torch/ssd/transforms.py, torch/ssd/inspect.py]` ? What would make more sense to a Rikai contributor adding a new model?

2. I forgot that we have both `rikai/contrib` and `rikai/python/rikai/contrib`. What goes where? Is there a reason to keep maintaining both?

Reply:
> I forgot that we have both rikai/contrib and rikai/python/rikai/contrib. What goes where? Is there a reason to keep maintaining both?

So for `rikai/python/rikai/contrib` currently contains torch/tf code, the dependencies have been installed via rikai's setup.py already.

The ones under `rikai/contrib` are the ones have separate dependencies, and less common than tf/torch.

> I wonder if it's better to organize things as [torch/transforms/ssd.py, torch/inspect/ssd.py] or [torch/ssd/transforms.py, torch/ssd/inspect.py]? What would make more sense to a Rikai contributor adding a new model?

I don't have a good answer for now.

Reply:
So it goes down whether the `contrib` code will be organized by functionality or model?

For example, in a model centric view

```
torch/ssd.py.  # valina ssd
torch/ssd_class_scores_extractor.py.  # for this PR
torch/ssd_some_other_feature_extractor.py.   
torch/faster_crnn.py
```

Each file / module export `SCHEMA / pre_processing / post_processing` ?

wdyt @changhiskhan 


A side note: i've tried to add a mixin to `torch.nn.Module`

```
class SSDClassScoreExtractor(torch.nn.Module,  RikaiModule):
    ...
```

However this mixin will be erased after convert the model to torchscript.

Reply:
> So for rikai/python/rikai/contrib currently contains torch/tf code, the dependencies have been installed via rikai's setup.py already.

We don't have to deal with it in this PR, but then should we just put `rikai/python/rikai/contrib/torch*` into `rikai/python/torch`? And like should `rikai/python/rikai/contrib/coco` just get moved to `rikai/contrib/coco` ? My main concern is that if it's confusing for me, newcomers will have no hope of making sense of the difference


> So it goes down whether the contrib code will be organized by functionality or model?

yeah exactly.

> torchscript

that's silly. torchscript is silly